Py3: remove __getslice__ method

bastiak opened the pull request:
__getslice__ is deprecated in py2 and it is not used in py3 at all.
Instead of this __getitem__ with slice() index should be used.

Please note that WireData class must still use __getslice__, because it
inherites from 'binary_type' object that has implemented __getslice__
thus this method has to overriden in WireData class.

bastiak commented:
Fixes: #180

rthalley commented:
We need to do a bit more, because just applying the patch weakens WireData back into ordinary slice semantics on both python 2 and 3.  (3 has already been weakened.). Python slices are sloppy, in that you can specify upper bounds outside the range, e.g. foo[1:99] on something that is only 3 items big is not an error.  Dnspython wants this to be an error, which is why we have an augmented class.

The right thing is to have the __getitem__ check for a slice, and if so look a the slices indices() and do the getitems that the existing wiredata does.

bastiak commented:
It was hard, but I fixed it, frankly speaking I don't like this violation of default python behavior. However it is backward compatible. I added tests to test this custom behavior. Please check license file in the new test file if it is OK.

bastiak commented:
Any update to this PR?
